roles/ruby_s/tasks/main.yml

```diff
@@ -0,0 +1,65 @@
+---
+- name: ruby | check version
+  ansible.builtin.shell: ruby -v | awk '{print $2}'
+  register: installed_ruby
+  ignore_errors: true
+
+- name: prints out the installed_ruby variables
+  ansible.builtin.debug:
+    msg: ruby version "{{ installed_ruby.stdout }}"
+
+- include_tasks: cleanup.yml
+  when: installed_ruby.stdout != "{{ desired_ruby_version }}"
+
+- name: ruby |lookup download path and checksum from ruby-lang release index
+  ansible.builtin.shell: curl https://cache.ruby-lang.org/pub/ruby/index.txt | grep {{ ruby_version }}.tar.gz
```

Comment:
Could we use the `get_url` module instead? I'm not sure how to replicate the grep part, so maybe not, but it would be worth exploring.